### ISSUE TITLE
Abnormal error

### ISSUE THREAD

**Issue by ewoutcrombez** [marbl/canu#2322]
Hello,

I tried running Canu using the command `canu -p Canu -d ../results/Canu genomeSize=158m corOutCoverage=200 correctedErrorRate=0.15 -nanopore reads.fastq.gz` on a slurm grid cluster, but got this "abnormal" error:
```
ERROR:
ERROR:  Failed with exit code 139.  (rc=35584)
ERROR:

CRASH:
CRASH: canu 2.2
CRASH: Please panic, this is abnormal.
CRASH:
CRASH:   splitReads failed.
CRASH:
CRASH: Failed at /shared/clssoft/apps/x86_64/canu/2.2/bin/../lib/site_perl/canu/OverlapBasedTrimming.pm line 140.
CRASH: 	canu::OverlapBasedTrimming::splitReads("SP0407_Canu") called at /software/shared/apps/x86_64/canu/2.2/bin/canu line 1106
CRASH: 
CRASH: Last 50 lines of the relevant log file (trimming/3-overlapbasedtrimming/SP0407_Canu.2.splitReads.err):
CRASH:
CRASH: Processing from ID 1 to 3024626 out of 3024626 reads, using errorRate = 0.15
CRASH: splitReads: stores/sqStore.H:403: uint32 sqStore::sqStore_getReadLength(uint32, sqRead_which): Assertion `id > 0' failed.
CRASH: 
CRASH: Failed with 'Aborted'; backtrace (libbacktrace):
CRASH: 
CRASH: Failed with 'Segmentation fault'; backtrace (libbacktrace):
CRASH:
```
I have no idea what is the cause of this error. Could you help me resolve this? Many thanks!

**Comment by rotifergirl** [comment]
Hi! I'm getting a very similar error, but at a different step.

The log when I run my canu command ends with:

```
-- Starting command on Sun Jul 14 15:13:23 2024 with 436487.224 GB free disk space

    cd correction
    /powerplant/workspace/appscratch/miniconda/cfnjxb_canu/bin/generateCorrectionLayouts \
      -S ../SNA_V2_canu.seqStore \
      -O  ./SNA_V2_canu.ovlStore \
      -C  ./SNA_V2_canu.corStore.WORKING \
      -eC 80 \
      -xC 40 \
    > ./SNA_V2_canu.corStore.err 2>&1

-- Finished on Sun Jul 14 16:43:22 2024 (5399 seconds) with 436133.289 GB free disk space
----------------------------------------

ERROR:
ERROR:  Failed with exit code 139.  (rc=35584)
ERROR:

ABORT:
ABORT: canu 2.2
ABORT: Don't panic, but a mostly harmless error occurred and Canu stopped.
ABORT: Try restarting.  If that doesn't work, ask for help.
ABORT:
ABORT:   failed to generate correction layouts.
ABORT:
ABORT: Disk space available:  436133.289 GB
ABORT:
ABORT: Last 50 lines of the relevant log file (correction/SNA_V2_canu.corStore.err):
ABORT:
ABORT:   generateCorrectionLayouts: stores/sqStore.H:403: uint32 sqStore::sqStore_getReadLength(uint32, sqRead_which): Assertion `id > 0' failed.
ABORT:   
ABORT:   Failed with 'Aborted'; backtrace (libbacktrace):
ABORT:   
ABORT:   Failed with 'Segmentation fault'; backtrace (libbacktrace):
ABORT:
```

I have also tried runnging the generateCorrectionLayouts command with 64GB RAM, but after some time, I get the same segmentation fault

**Comment by brianwalenz** [comment]
Not sure what could be causing this, but I got rid of the assert() statement that was directly causing the crash.  This probably means that it will just fail somewhere else.  I'd need at least a stack trace (`make BUILDSTACKTRACE=1`), and probably even an actual crashing example, to do more.
